pyomo/util/tests/test_latex_printer.py

```diff
@@ -0,0 +1,349 @@
+    m.constraint_5 = pe.Constraint(expr=Expr_if(m.x <= 1.0, m.z, m.y) <= 1.0)
+
+    def blackbox(a, b):
+        return sin(a - b)
+
+    m.bb = ExternalFunction(blackbox)
+    m.constraint_6 = pe.Constraint(expr=m.x + m.bb(m.x, m.y) == 2)
+
+    m.I = pe.Set(initialize=[1, 2, 3, 4, 5])
+    m.J = pe.Set(initialize=[1, 2, 3])
+    m.K = pe.Set(initialize=[1, 3, 5])
+    m.u = pe.Var(m.I * m.I)
+    m.v = pe.Var(m.I)
+    m.w = pe.Var(m.J)
+    m.p = pe.Var(m.K)
+
+    m.express = pe.Expression(expr=m.x**2 + m.y**2)
+
+    def ruleMaker(m, j):
+        return (m.x + m.y) * sum(m.v[i] + m.u[i, j] ** 2 for i in m.I) <= 0
+
+    m.constraint_7 = pe.Constraint(m.I, rule=ruleMaker)
+
+    def ruleMaker(m):
+        return sum(m.p[k] for k in m.K) == 1
+
+    m.constraint_8 = pe.Constraint(rule=ruleMaker)
+
+    def ruleMaker(m):
+        return (m.x + m.y) * sum(m.w[j] for j in m.J)
+
+    m.objective_2 = pe.Objective(rule=ruleMaker)
+
+    m.objective_3 = pe.Objective(expr=m.x + m.y + m.z, sense=-1)
+
+    return m
+
+
+def generate_simple_model():
+    import pyomo.environ as pe
+
+    m = pe.ConcreteModel(name='basicFormulation')
+    m.x = pe.Var()
+    m.y = pe.Var()
+    m.objective_1 = pe.Objective(expr=m.x + m.y)
+    m.constraint_1 = pe.Constraint(expr=m.x**2 + m.y**2.0 <= 1.0)
+    m.constraint_2 = pe.Constraint(expr=m.x >= 0.0)
+
+    m.I = pe.Set(initialize=[1, 2, 3, 4, 5])
+    m.J = pe.Set(initialize=[1, 2, 3])
+    m.K = pe.Set(initialize=[1, 3, 5])
+    m.u = pe.Var(m.I * m.I)
+    m.v = pe.Var(m.I)
+    m.w = pe.Var(m.J)
+    m.p = pe.Var(m.K)
+
+    def ruleMaker(m, j):
+        return (m.x + m.y) * sum(m.v[i] + m.u[i, j] ** 2 for i in m.I) <= 0
+
+    m.constraint_7 = pe.Constraint(m.I, rule=ruleMaker)
+
+    def ruleMaker(m):
+        return sum(m.p[k] for k in m.K) == 1
+
+    m.constraint_8 = pe.Constraint(rule=ruleMaker)
+
+    return m
+
+
+def generate_simple_model_2():
+    import pyomo.environ as pe
+
+    m = pe.ConcreteModel(name='basicFormulation')
+    m.x_dot = pe.Var()
+    m.x_bar = pe.Var()
+    m.x_star = pe.Var()
+    m.x_hat = pe.Var()
+    m.x_hat_1 = pe.Var()
+    m.y_sub1_sub2_sub3 = pe.Var()
+    m.objective_1 = pe.Objective(expr=m.y_sub1_sub2_sub3)
+    m.constraint_1 = pe.Constraint(
+        expr=(m.x_dot + m.x_bar + m.x_star + m.x_hat + m.x_hat_1) ** 2
+        <= m.y_sub1_sub2_sub3
+    )
+    m.constraint_2 = pe.Constraint(
+        expr=(m.x_dot + m.x_bar) ** -(m.x_star + m.x_hat) <= m.y_sub1_sub2_sub3
+    )
+    m.constraint_3 = pe.Constraint(
+        expr=-(m.x_dot + m.x_bar) + -(m.x_star + m.x_hat) <= m.y_sub1_sub2_sub3
+    )
+
+    return m
+
+
+class TestLatexPrinter(unittest.TestCase):
+    def test_latexPrinter_objective(self):
+        m = generate_model()
+        pstr = latex_printer(m.objective_1)
+        bstr = ''
+        bstr += '\\begin{equation} \n'
+        bstr += '    & \\text{minimize} \n'
+        bstr += '    & & x + y + z \n'
+        bstr += '\end{equation} \n'
```

Comment:
You can do all of these as a multi-line string:
```python
bstr = """
\\begin{equation} \n
    & \\text{minimize} \n
....

"""
```

Reply:
Fixed